File: 383-ransom-note/383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) 
    {
        int[] r = new int[26];
        int[] m = new int[26];
        for(int i=0; i<ransomNote.length(); i++)
            r[ransomNote.charAt(i)-'0'-49]++;
        for(int i=0; i<magazine.length(); i++)
            m[magazine.charAt(i)-'0'-49]++;
        for(int i=0; i<26; i++)
            if(m[i]<r[i])
                return false;
        return true;
    }
}

File: 383-ransom-note/383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) 
    {
        int r[26]={0}, m[26]={0};
        for(char ch:ransomNote)
            r[ch-'a']++;
        for(char ch:magazine)
            m[ch-'a']++;
        for(int i=0; i<26; i++)
            if(m[i]<r[i] && r[i]>0)
                return false;
        return true;
    }
};